sysl2/sysl/ints.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"flag"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+func GenerateIntegrations(
+	root_model, title, output, project, filter, modules string,
+	exclude []string, clustered, epa bool,
+) map[string]string {
+	r := make(map[string]string)
+	mod := loadApp(root_model, modules)
+
+	if len(exclude) == 0 && project != "" {
+		exclude = append(exclude, project)
+	}
+	excludeStrSet := MakeStrSet(exclude...)
+	ds := NewDependencySet()
+	ds.CollectAppDependencies(mod)
+
+	// The "project" app that specifies the required view of the integration
+	app := mod.GetApps()[project]
+	of := MakeFormatParser(output)
+	// Interate over each endpoint within the selected project
+	for epname, endpt := range app.GetEndpoints() {
+		// build the set of excluded items
+		excludes := MakeStrSetFromSpecificAttr("exclude", endpt.GetAttrs())
+		passthroughs := MakeStrSetFromSpecificAttr("passthrough", endpt.GetAttrs())
+		// endpt.stmt's "action" will conatain the "apps" whose integration is to be drawn
+		// each one of these will be placed into the "integration" list
+		integrations := MakeStrSetFromActionStatement(endpt.GetStmt())
+
+		arr := ds.ToSlice()
+		highlights := FindApps(mod, excludeStrSet, integrations, arr, true)
+		apps := FindApps(mod, excludeStrSet, highlights, arr, false)
+		apps = apps.Difference(excludes)
```

Comment:
iterating TWICE on  this CALL LIST (by FindApps).

Reply:
Removed the first FindApps() invocation as the logic is just check the integrations should be highlighted. The second FindApps() is to find all the relevant apps in the final integration view.